x-pack/plugins/security_solution/scripts/run_cypress/parallel.ts

```diff
@@ -204,69 +204,72 @@ ${JSON.stringify(cypressConfigFile, null, 2)}
         _.pull(fleetServerPorts, fleetServerPort);
       };
 
-      await pMap(
-        files,
-        async (filePath) => {
-          let result:
-            | CypressCommandLine.CypressRunResult
-            | CypressCommandLine.CypressFailedRunResult
-            | undefined;
-          await withProcRunner(log, async (procs) => {
-            const abortCtrl = new AbortController();
-
-            const onEarlyExit = (msg: string) => {
-              log.error(msg);
-              abortCtrl.abort();
-            };
-
-            const esPort: number = getEsPort();
-            const kibanaPort: number = getKibanaPort();
-            const fleetServerPort: number = getFleetServerPort();
-            const specFileFTRConfig = parseTestFileConfig(filePath);
-            const ftrConfigFilePath = path.resolve(
-              _.isArray(argv.ftrConfigFile) ? _.last(argv.ftrConfigFile) : argv.ftrConfigFile
-            );
-
-            const config = await getFTRConfig({
-              log,
-              esPort,
-              kibanaPort,
-              fleetServerPort,
-              ftrConfigFilePath,
-              specFilePath: filePath,
-              specFileFTRConfig,
-              isOpen,
-            });
-
-            const createUrlFromFtrConfig = (
-              type: 'elasticsearch' | 'kibana' | 'fleetserver',
-              withAuth: boolean = false
-            ): string => {
-              const getKeyPath = (keyPath: string = ''): string => {
-                return `servers.${type}${keyPath ? `.${keyPath}` : ''}`;
+      const failedSpecFilePaths: string[] = [];
```

Comment:
If we want to reuse the `runSpecs` method then `failedSpecFilePaths` needs to live outside of that. It's not going to end up in the loop, because when we execute it for the 2nd time we create a new array with the values from the first run, so even if we keep pushing failing tests to the `failedSpecFilePaths` in 2nd run it doesn't affect the values passed to 2nd run

Reply:
Wonderful, thanks!